WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -1161,13 +1161,29 @@ private extension ProductsViewController {
     }
 
     func configureLeftBarBarButtomItemAsScanningButtonIfApplicable() {
-        guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.scanToUpdateInventory),
-              UIImagePickerController.isSourceTypeAvailable(.camera) else {
-            navigationItem.leftBarButtonItem = nil
-            return
+        // The feature breaks if the Square plugin is active, since modifies inventory management logic
+        // If the plugin is active, we'll hide the inventory scanner button
+        // More details: https://wp.me/pdfdoF-2Nq
+        isPluginActive(SitePlugin.SupportedPlugin.square) { result in
+            switch result {
+            case true:
+                self.navigationItem.leftBarButtonItem = nil
+            case false:
+                guard ServiceLocator.featureFlagService.isFeatureFlagEnabled(.scanToUpdateInventory),
+                      UIImagePickerController.isSourceTypeAvailable(.camera) else {
+                    self.navigationItem.leftBarButtonItem = nil
+                    return
+                }
+                self.navigationItem.leftBarButtonItem = self.createAddOrderByProductScanningButtonItem()
+            }
         }
+    }
 
-        navigationItem.leftBarButtonItem = createAddOrderByProductScanningButtonItem()
+    func isPluginActive(_ plugin: String, completion: @escaping (Bool) -> (Void)) {
+        let action = SystemStatusAction.fetchSystemPluginListWithNameList(siteID: siteID, systemPluginNameList: [plugin]) { plugin in
+            completion(plugin?.active == true)
+        }
+        ServiceLocator.stores.dispatch(action)
```

Comment:
As a side note of using `ServiceLocator` here, I'm calling it directly for consistency with the rest of the class, however, there's 35 instances of `ServiceLocator` in this file used across multiple features and would be great to inject it instead. 

Making the change here felt out of scope for this PR, I will open a different PR in order to make the changes after we've wrapped up this project, so I can test that everything still works as expected.